internal/app/summon/command/temp_factory_test.go

```diff
@@ -0,0 +1,99 @@
+package command
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+// TODO: Test TempFactory.Cleanup()
+// TODO: Test TempFactory.Push()
+
+func splitEq(s string) (string, string) {
+	a := strings.SplitN(s, "=", 2)
+	return a[0], a[1]
+}
+
+type envSnapshot struct {
+	env []string
+}
+
+func ClearEnv() *envSnapshot {
+	e := os.Environ()
+
+	for _, s := range e {
+		k, _ := splitEq(s)
+		os.Setenv(k, "")
+	}
+	return &envSnapshot{env: e}
+}
+
+func (e *envSnapshot) RestoreEnv() {
+	ClearEnv()
+	for _, s := range e.env {
+		k, v := splitEq(s)
+		os.Setenv(k, v)
+	}
+}
+
+func TestTempFactory_NewTempFactory(t *testing.T) {
+	Convey("Uses constructor arg path if provided", t, func() {
+		tempFactory := NewTempFactory("somedir")
+
+		So(tempFactory, ShouldResemble, TempFactory{
+			files: []string(nil),
+			path:  "somedir",
+		})
+	})
+
+	Convey("When constructor path is not provided", t, func() {
+		env := ClearEnv()
+		defer env.RestoreEnv()
+
+		Convey("tries using DEVSHM first", func() {
+			tempFactory := NewTempFactory("")
+
+			So(tempFactory, ShouldResemble, TempFactory{
+				files: []string(nil),
+				path:  DEVSHM,
+			})
+		})
+
+		Convey("tries using homedir prefix if DEVSHM is not available", func() {
+			// Create a fake $HOME
+			home, err := ioutil.TempDir("", "secretless_test")
+			So(err, ShouldBeNil)
+
+			defer func() {
+				os.RemoveAll(home)
+			}()
+
+			os.Setenv("HOME", home)
+
+			// Override DEVSHM
+			devshm := DEVSHM
+			DEVSHM = ""
+			defer func() {
+				DEVSHM = devshm
+			}()
+
+			tempFactory := NewTempFactory("")
+			So(tempFactory.path, ShouldStartWith, home)
+		})
+
+		Convey("tries using os.TempDir as last resort", func() {
+			// Override DEVSHM
+			devshm := DEVSHM
+			DEVSHM = ""
+			defer func() {
+				DEVSHM = devshm
+			}()
+
```

Comment:
Per my other comment, making a const into a var so we can fiddle with it in tests feels hacky.  What this is telling us that `os.Stat` is a dependency.  So you can inject it (variadic args allow you to do this without messing up the current ctor interface).  Then in your tests you can inject a double that will return a mock `fi` object, allowing you to do all this cleanly.

Reply:
You can resolve this.  New tests look good 👍